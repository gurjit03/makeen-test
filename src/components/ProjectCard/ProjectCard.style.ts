import styled from 'styled-components';
import { colors, fontSizes, verticalSpacing } from '../../config/jss';

export const Wrapper = styled.article`
    padding: ${verticalSpacing * 2}px;
    border-radius: ${verticalSpacing * 2}px;
    border: 2px solid ${colors.accent};
    background-color: ${colors.white};
    max-width: 200px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-items: center;
`;

export const ProjectDuration = styled.div`
    display: flex;
    width: 100%;
    align-items: center;
    justify-content: space-evenly;
    margin-bottom: ${verticalSpacing * 2}px;
`;

export const ProjectName = styled.h2`
    font-size: ${fontSizes.large}px;
    margin: 0;
    margin-bottom: ${verticalSpacing}px;
`;

export const ProjectStatus = styled.p<{color: string}>`
    color: ${({color}) => color}
    font-weight: 800;
    margin-bottom: ${verticalSpacing * 2}px;
`;

export const ProjectEndDuration = styled.p`
    font-weight: 800;
`;

export const ProjectAction = styled.div`
    width: 100%;
    display: flex;
    justify-content: flex-end;
    margin-top: ${verticalSpacing * 2}px;
`;