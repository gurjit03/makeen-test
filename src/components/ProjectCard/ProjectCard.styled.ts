import styled from 'styled-components';
import { colors, standardSizes, verticalSpacing } from '../../config/jss';

export const Wrapper = styled.article`
    padding: ${verticalSpacing}px;
    border-radius: ${verticalSpacing * 2}px;
    border: 2px solid ${colors.accent};
`;

export const ProjectName = styled.h2`
    font-size: ${standardSizes.large}px;
    margin: 0;
    margin-bottom: ${verticalSpacing}px;
`;

export const ProjectStatus = styled.p<{color: string}>`
    color: ${({color}) => color}
    font-weight: 800;
`;

export const ProjectEndDuration = styled.p`
    font-weight: 800;
`;