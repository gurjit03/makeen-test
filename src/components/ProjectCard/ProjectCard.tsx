
import React from 'react';
import { Wrapper, ProjectName, ProjectStatus, ProjectEndDuration, ProjectDuration, ProjectAction } from './ProjectCard.style';
import { Project as ProjectProps } from '../../interfaces/Project.interface';
import { Button } from '../../ui';
import { getColorsForStatus, getProperDate } from '../../utils';

interface ProjectCardProps extends ProjectProps {
    onActionClick: () => void
}

const ProjectCard: React.FC<ProjectCardProps> = (props) => {
    const {
        id,
        name,
        status,
        duration,
        description,
        onActionClick
    } = props;
    const projectStatusColor = getColorsForStatus(status);

    return (
        <Wrapper>
            <ProjectName>{name}</ProjectName>
            <ProjectStatus color={projectStatusColor}>{status}</ProjectStatus>
            <ProjectDuration>
                <p>{getProperDate(duration.start)}</p> to <ProjectEndDuration>{getProperDate(duration.end)}</ProjectEndDuration>
            </ProjectDuration>
            <p>{description.substr(0, 120)}</p>
            <ProjectAction>
                <Button variant="outlined" color="primary" onClick={onActionClick}>
                    view
                </Button>
            </ProjectAction>
        </Wrapper>
    )
}

export default ProjectCard