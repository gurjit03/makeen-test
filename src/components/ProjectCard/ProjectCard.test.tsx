import React from 'react'
import { render, screen } from '@testing-library/react'
import ProjectCard from './ProjectCard'

test('renders project card properly', () => {
    render(<ProjectCard />)
    const linkElement = screen.getByText(/learn react/i)
    expect(linkElement).toBeInTheDocument()
})
