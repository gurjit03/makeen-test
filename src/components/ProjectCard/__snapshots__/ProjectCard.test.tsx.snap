// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectCard /> should render properly 1`] = `
<DocumentFragment>
  <article
    class="sc-bdfBQB floGhh"
  >
    <h2
      class="sc-dlfnuX eimhgL"
    >
      john doe
    </h2>
    <p
      class="sc-hKgJUU SMLVg"
      color="#3CA6E6"
    >
      running
    </p>
    <div
      class="sc-gsTEea ezBYqA"
    >
      <p>
        11.9.21
      </p>
       to 
      <p
        class="sc-jSgvzq gbgGMA"
      >
        11.9.21
      </p>
    </div>
    <p
      class="sc-eCstlR gWgnLP"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <div
      class="sc-gKseQn laNytO"
    >
      <button
        class="sc-iBPTik bDICib"
        color="primary"
      >
        view
      </button>
    </div>
  </article>
</DocumentFragment>
`;
